docs/docs/gatsby-repl.md

```diff
@@ -1,9 +1,215 @@
 ---
 title: Gatsby REPL
-issue: https://github.com/gatsbyjs/gatsby/issues/7484
 ---
 
-This is a stub. Help our community expand it.
+The Gatsby REPL ("read-eval-print loop") is available via the Gatsby CLI by running the command `gatsby repl`. This gives you access to an interactive REPL shell within the context of your Gatsby environment. It can be used to retrieve general data and programmatically interact with it. If you have an idea for a new command, feel free to submit a PR for it!
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+This doc will give a brief description of each REPL command, expected output, and an example of what you can do with the command to digest the data. Examples are using the [Gatsby Starter Blog](/starters/gatsbyjs/gatsby-starter-blog/) as an demo environment, since it is currently the highest rated starter, and provides standard output for most of these commands.
+
+To get started, in your terminal, after running the initial site setup steps [here](/docs/quick-start), run the command `gatsby repl` to enter the interactive shell. If you are writing out a function, you can write that over multiple lines, as long as you do not use a semicolon or close a parenthesis or brace prematurely. This is helpful for running graphql queries and callback functions.
+
+## REPL Commands
+
+- [`babelrc`](#babelrc)
+- [`components`](#components)
+- [`getNode`](#getNode)
+- [`getNodes`](#getNodes)
+- [`nodes`](#nodes)
+- [`pages`](#pages)
+- [`schema`](#schema)
+- [`siteConfig`](#siteConfig)
+- [`staticQueries`](#staticQueries)
+
+### `babelrc`
+
+Returns an object with the global `babelrc` settings.
+
+Usage: `babelrc`
+
+Example:
+
+```js
+// Command:
+gatsby > babelrc
+// Returns:
+{ stages:
+   { develop: { plugins: [], presets: [], options: [Object] },
+     'develop-html': { plugins: [], presets: [], options: [Object] },
+     'build-html': { plugins: [], presets: [], options: [Object] },
+     'build-javascript': { plugins: [], presets: [], options: [Object] } } }
+```
+
+### `components`
+
+Returns a Map object with all of the components in your Gatsby environment (see [Mozilla Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) docs for more information on Map objects and how to use them). Propeties that get returned: `name`, `componentPath`, `id`, `query`, and `hash`.
```

Comment:
also property `pages` (used in the code snippet below)?

Reply:
Thanks for pointing that out! I changed this in the most recent commit.